File: abc/333/c.py
def main():
    n = input()
    numbers = ["1", "1", "1"]
    repunits = []

    for i in range(333):
        repunits.append(int(numbers[0]) + int(numbers[1]) + int(numbers[2]))
        # print(f"{i}, {numbers[0]}, {numbers[1]}, {numbers[2]}")
        if (numbers[0] == numbers[1] == numbers[2]):
            numbers[0] += "1"
            numbers[1] = "1"
            numbers[2] = "1"
        elif (len(numbers[1]) == len(numbers[2])):
            numbers[1] += "1"
            numbers[2] = "1"
        else:
            numbers[2] += "1"

    print(repunits[int(n) - 1])

if __name__ == "__main__":
    main()

File: abc/333/c.cpp
#include <bits/stdc++.h>

#define ll long long
#define ull unsigned long long

int main() {
    int n;

    std::cin >> n;

    std::vector<std::string> numbers = {"1", "1", "1"};
    std::vector<ull> repunits = {};

    for (int i = 0; i < 333; i++) {
        int index = i % 3;
        repunits.push_back(std::stoull(numbers[0]) + std::stoull(numbers[1]) +
                           std::stoull(numbers[2]));
        std::cout << i << std::endl;
        std::cout << numbers[0] << ", " << numbers[1] << ", " << numbers[2]
                  << std::endl;
        numbers[index] += "1";
    }

    std::cout << repunits[n] << std::endl;

    return 0;
}
